Lib/test/test_marshal.py

```diff
@@ -344,6 +344,20 @@ def test_eof(self):
         for i in range(len(data)):
             self.assertRaises(EOFError, marshal.loads, data[0: i])
 
+    def test_deterministic_sets(self):
+        # bpo-37596: sets and frozensets must always marshal deterministically!
```

Comment:
The word "deterministic" is correct but doesn't help motivate the algorithm to someone who hasn't read the tracker issue.  I recommend:

> To support reproducible builds, sets and frozensets need to be stored with the keys in consistent order even when they have been scrambled by hash randomization.  Here we use an order equivalent to:  sorted(sorted(s, key=marshal.dumps)).  See bpo-37596 for details.